parquet/schema.go

```diff
@@ -0,0 +1,122 @@
+package parquet
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/segmentio/parquet-go"
+)
+
+func (c *Client) makeStruct(cols schema.ColumnList) any {
+	sf := make([]reflect.StructField, len(cols))
+	for i := range cols {
+		sf[i] = c.structFieldFromColumn(i, cols[i])
+	}
+
+	return reflect.New(reflect.StructOf(sf)).Elem().Interface()
+}
+
+func (c *Client) structFieldFromColumn(index int, col schema.Column) reflect.StructField {
+	el := schemaTypeToGoType(col.Type)
+	f := reflect.StructField{
+		Name: formatFieldNameWithIndex(index),
+		Type: reflect.TypeOf(el),
+	}
+
+	tg := `parquet:"` + col.Name
+	if opts := c.structOptsForColumn(col); len(opts) > 0 {
+		tg += "," + strings.Join(opts, ",")
+	}
+	tg += `"`
+
+	f.Tag = reflect.StructTag(tg)
+	return f
+}
+
+func (c *Client) structOptsForColumn(col schema.Column) []string {
+	opts := []string{c.spec.Compression}
+
+	switch col.Type {
+	case schema.TypeJSON:
+		opts = append(opts, "json")
+	case schema.TypeUUID:
+		opts = append(opts, "uuid")
+	case schema.TypeTimestamp:
+		opts = append(opts, "timestamp")
+	case schema.TypeInt, schema.TypeString, schema.TypeByteArray:
+		opts = append(opts, "delta")
+	case schema.TypeStringArray, schema.TypeIntArray, schema.TypeUUIDArray, schema.TypeCIDRArray, schema.TypeInetArray:
+		opts = append(opts, "list")
```

Comment:
Setting this is supposed to make lists 'iterable' otherwise they're all show up as byte arrays...